tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+                            }
+                        ]
+                    },
+                    "Parameters": {
+                        "classification": "json",
+                        "compressionType": "none",
+                        "typeOfData": "file",
+                    }
+                }
+        )
+        return glue_table
+
+    @pytest.fixture(scope="class")
+    def glue_flat_table_location(self, glue_database_path, glue_flat_table_name):
+        glue_flat_table_location = os.path.join(
+                glue_database_path,
+                glue_flat_table_name.replace("_", "=", 1)
+            ) + "/"
+        return glue_flat_table_location
+
+    @pytest.fixture(scope="class")
+    def glue_flat_table(self, glue_database_name, glue_database_path,
+                        glue_flat_table_name, glue_flat_table_location):
+        glue_client = boto3.client("glue")
+        glue_table = glue_client.create_table(
+                DatabaseName=glue_database_name,
+                TableInput={
+                    "Name": glue_flat_table_name,
+                    "Description": "A table for pytest unit tests.",
+                    "Retention": 0,
+                    "TableType": "EXTERNAL_TABLE",
+                    "StorageDescriptor": {
+                        "Location": glue_flat_table_location,
+                        "InputFormat": "org.apache.hadoop.mapred.TextInputFormat",
+                        "OutputFormat": "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat",
+                        "Compressed": False,
+                        "StoredAsSubDirectories": False,
+                        "Columns": [
+                            {
+                                "Name": "GlobalKey",
+                                "Type": "string"
+                            },
+                            {
+                                "Name": "export_end_date",
+                                "Type": "string"
+                            }
+                        ]
+                    },
+                    "Parameters": {
+                        "classification": "json",
+                        "compressionType": "none",
+                        "typeOfData": "file",
+                    }
+                }
+        )
+        return glue_table
+
+    def upload_test_data_to_s3(self, path, s3_bucket, table_location):
+        # TODO: Get this working and use in `glue_test_data` for all 4 test data
+        s3_client = boto3.client("s3")
+        file_basename = os.path.basename(path)
+        s3_key = os.path.relpath(
+                os.path.join(table_location, file_basename),
+                f"s3://{s3_bucket}"
+        )
+        s3_client.upload_file(
+                Filename=path,
+                Bucket=s3_bucket,
+                Key=s3_key
+        )
+
+    @pytest.fixture()
+    def glue_test_data(self, glue_flat_table_location, glue_nested_table_location,
```

Comment:
What are you trying to achieve here?  This seems like a job for `setup_method` for the test to execute this chunk of code for all the tests in the beginning.

Reply:
I'm using fixtures rather than xunit-style `setup_method`. I had to navigate a lot of complexity here because we want to run these fixtures once per class (like `setup_method`) but we also wanted to use the other fixtures which these "setup" fixtures are dependent on in different tests. I can't pass a fixture to `setup_method`, so using `setup_method` would require maintaining a lot of redundant set-up logic.

Unfortunately, one of the third-party fixtures we use (`datadir`) is scoped at the function level, and function-level fixtures cannot be used in what ought to be class-level fixtures (e.g., `glue_test_data`, `glue_crawler`), so to get around this these fixtures are downgraded to function-level fixtures and each are passed to their own test at the beginning of the class (`test_upload_test_data` and `test_run_glue_crawler`), which is functionally equivalent to `setup_method` but gives us the advantage of being able to use reusable fixtures in these "setup" fixtures.

Reply:
Ah, I see. Thanks for the explanation!